inference-engine/tests/functional/plugin/gna/pass_tests/layers_restrictions.cpp

```diff
@@ -50,6 +50,32 @@ struct FullyConnectedBatchSizeLessThanOrEqual8 {
     }
 };
 
+struct ConcatInHorizontalAxis {
+    static const char* getName() { return "ConcatInHorizontalAxis"; }
+    static std::shared_ptr<ngraph::Function> createTopology(const InferenceEngine::Precision& netPrecision) {
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        std::vector<std::vector<size_t>> inputShapes = {{1, 2, 3, 1}, {1, 2, 2, 1}};
```

Comment:
I'd check more possible cases here:
- 4d/ 3d /2d
- different axes for concatenation
- different layers before concat (at least convolutions in different layouts and matmuls)
- different input shapes, including cases when batch size > 1

I'll try to describe here different cases which should be supported as I understand them (please, correct me):
1. Convolution -> Concat
support concatenation in axis 0 or dimension before which all dimensions are 1 (if look at dimensions in NHWC layout)
2. MatMul -> Concat
we should move batch size to the end of dimensions (since data is transposed relative to its original order reflected in dims)
for the modified order support concatenation  in axis 0 or later dimension before which all dimensions are 1